src/Common/src/Interop/Windows/NtDll/Interop.NtQueryInformationProcess.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class NtDll
+    {
+        [DllImport(Libraries.NtDll, CharSet = CharSet.Unicode)]
+        internal static extern unsafe uint NtQueryInformationProcess(SafeProcessHandle ProcessHandle, PROCESSINFOCLASS ProcessInformationClass, void* ProcessInformation, uint ProcessInformationLength, out uint ReturnLength);
+
+        internal enum PROCESSINFOCLASS : int
+        {
+            ProcessBasicInformation = 0,
+            ProcessDebugPort = 7,
+            ProcessWow64Information = 26,
+            ProcessImageFileName = 27,
+            ProcessBreakOnTermination = 29,
+            ProcessSubsystemInformation = 75
+        };
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct PROCESS_BASIC_INFORMATION
```

Comment:
Nit, although I see pinvoke.net somehow shows all the members, some of these are not officially documented. The public headers define it as 
```
typedef struct _PROCESS_BASIC_INFORMATION {
    PVOID Reserved1;
    PPEB PebBaseAddress;
    PVOID Reserved2[2];
    ULONG_PTR UniqueProcessId;
    PVOID Reserved3;
} PROCESS_BASIC_INFORMATION;
```
We try to match public headers as closely as possible, so we would have
```
        public IntPtr Reserved1; // or UIntPtr
        public IntPtr PebBaseAddress;
        public IntPtr Reserved2a;
        public IntPtr Reserved2b;
        public IntPtr BasePriority;
        public UIntPtr UniqueProcessId;
        public UIntPtr Reserved3;
```
Some of this is a matter of taste I think, eg., `void*` would also work, but we don't seem to use that as much. I'm not sure of the general preference.

Reply:
I'll fix it.

Reply:
But we are relying on InheritedFromUniqueProcessId which is among the undocumented fields. Although, it is "documented" all over the web. It seems the documented way is to enumerate with the ToolHelp32 API's - ugh.

Reply:
Never mind, it is documented in the DDK